src/metabase/transforms/materialize.clj

```diff
@@ -0,0 +1,81 @@
+(ns metabase.transforms.materialize
+  (:require [metabase.api.common :as api]
+            [metabase.models
+             [card :as card :refer [Card]]
+             [collection :as collection]]
+            [metabase.query-processor.middleware
+             [add-implicit-clauses :as qp.imlicit-clauses]
+             [annotate :as qp.annotate]
+             [fetch-source-query :as qp.source-query]
+             [resolve-joins :as qp.resolve-joins]]
+            [metabase.util :as u]
+            [toucan.db :as db]))
+
+(defn infer-cols
+  "Infer column types from given (MBQL) query."
+  [query]
+  (-> query
+      (#'qp.source-query/resolve-all)
+      (#'qp.imlicit-clauses/maybe-add-implicit-clauses)
+      (#'qp.resolve-joins/resolve-joins*)
+      (#'qp.annotate/add-column-info* nil)
+      :cols))
+
+(declare get-or-create-root-container-collection!)
+
+(defn- root-container-location
+  []
+  (collection/children-location
+   (db/select-one ['Collection :location :id]
+     :id (get-or-create-root-container-collection!))))
+
+(defn get-collection
+  "Get collection named `collection-name`. If no location is given root collection for automatically
+   generated transforms is assumed (see `get-or-create-root-container-collection!`)."
+  ([collection-name]
+   (get-collection collection-name (root-container-location)))
+  ([collection-name location]
+   (db/select-one-id 'Collection
```

Comment:
quoting Toucan model names is only an option to prevent circular references between model namespaces. You should require the namespace and refer to `Collection` instead so things like `lein ring server` will have the correct dependency graph and work properly